File: hfsfolder-php/model/Folder.php
<?php


namespace model;

require_once(__ROOT__.'/model/PreFile.php');

use model\PreFile as PreFile;

class Folder extends PreFile
{
    private $code = 0;
    private $order = 0;
    private $parentCodFolder = 0;
    private $folderType = '';
    private $path = "";
    private $parentName = "";
    private $parentPath = "";

    public $separatorChar = '/';

    public function __construct($preFile) {
        $this->setName($preFile->getName());
        $this->setSize($preFile->getSize());
        $this->setModified($preFile->getModified());
        $this->setAttributes($preFile->getAttributes());
        $this->setFormatedSize($preFile->getFormatedSize());
        $this->setFormatedModified($preFile->getFormatedModified());
        $this->setOriginalPath($preFile->getOriginalPath());
    }

    /**
     * @param int $code
     */
    public function setCode($code)
    {
        $this->code = $code;
    }

    /**
     * @return int
     */
    public function getCode()
    {
        return $this->code;
    }

    /**
     * @param int $order
     */
    public function setOrder($order)
    {
        $this->order = $order;
    }

    /**
     * @return int
     */
    public function getOrder()
    {
        return $this->order;
    }

    /**
     * @return int
     */
    public function getParentCodFolder()
    {
        return $this->parentCodFolder;
    }

    /**
     * @param int $parentCodFolder
     */
    public function setParentCodFolder($parentCodFolder)
    {
        $this->parentCodFolder = $parentCodFolder;
    }

    /**
     * @return string
     */
    public function getFolderType()
    {
        return $this->folderType;
    }

    /**
     * @param string $folderType
     */
    public function setFolderType($folderType)
    {
        $this->folderType = $folderType;
    }

    /**
     * @return string
     */
    public function getPath()
    {
        return $this->path;
    }

    /**
     * @param string $path
     */
    public function setPath($path)
    {
        $this->path = $path;
    }

    /**
     * @return string
     */
    public function getParentName()
    {
        return $this->parentName;
    }

    /**
     * @param string $parentName
     */
    public function setParentName($parentName)
    {
        $this->parentName = $parentName;
    }

    /**
     * @return string
     */
    public function getParentPath()
    {
        return $this->parentPath;
    }

    /**
     * @param string $parentPath
     */
    public function setParentPath($parentPath)
    {
        $this->parentPath = $parentPath;
    }


    public function limparDados() {
        parent::limparDados();
        $this->code = 0;
        $this->order = 0;
        $this->parentCodFolder = 0;
        $this->folderType = 'A';
        $this->path = "";
        $this->parentName = "";
        $this->parentPath = "";
    }

    public function __toString() {
        return "Folder [codigo=" . $this->code . ", ordem=" . $this->order
            . ", codDirPai=" . $this->parentCodFolder . ", tipo=" . $this->folderType
            . ", caminho=" . $this->path . ", nomePai=" . $this->parentName
            . ", caminhoPai=" . $this->parentPath . "]";
    }

    public function toInsert($naba) {
        return "INSERT INTO Diretorios(aba, cod, ordem, nome, tam, tipo, modificado, atributos, coddirpai, caminho) VALUES("
            . $naba . "," . $this->code . "," . $this->order . ",'" . $this->getName() . "',"
            . $this->getSize() . ",'" . $this->folderType . "','" . $this->getFormatedModified()
            . "','" . $this->getAttributes() . "'," . $this->parentCodFolder . ",'" . $this->getPath()
            . "');";
    }

    public function toCVS() {
        return $this->code . ";" . $this->order . ";" . $this->getName()
            . ";" . $this->getSize() . ";" . $this->folderType . ";"
            . $this->getFormatedModified() . ";" . $this->getAttributes() . ";"
            . $this->parentCodFolder . ";" . $this->getPath();
    }

    public function toJSON(){
        return "{\n  \"name\" : \"" . $this->getName() . "\"," .
            "\n  \"size\" : " . $this->getSize() . "," .
            "\n  \"modified\" : " . $this->getModified() . "," .
            "\n  \"attributes\" : \"" . $this->getAttributes() . "\"," .
            "\n  \"formatedSize\" : \"" . $this->getFormatedSize() . "\"," .
            "\n  \"formatedModified\" : \"" . $this->getFormatedModified() . "\"," .
            "\n  \"code\" : " . $this->code . "," .
            "\n  \"order\" : " . $this->order . "," .
            "\n  \"parentCodFolder\" : " . $this->parentCodFolder . "," .
            "\n  \"folderType\" : \"" . $this->folderType . "\"," .
            "\n  \"path\" : \"" . $this->path . "\"," .
            "\n  \"parentName\" : \"" . $this->parentName . "\"," .
            "\n  \"parentPath\" : \"" . $this->parentPath . "\"," .
            "\n  \"originalPath\" : \"" . $this->getOriginalPath() . "\"\n}";
    }

}

File: hfsfolder-pascal/folder.pas
unit Folder;

{$mode objfpc}{$H+}

interface

uses
  Classes, SysUtils, PreFile;

type
  TFolder = class(TPreFile)
  private
    code: integer;
    order: integer;
    parentCodFolder: integer;
    folderType: char;
    path: string;
    parentName: string;
    parentPath: string;
  public
    separatorFile: char;

    constructor Create(prefile: TPreFile);
    destructor Destroy; override;

    function getCode(): integer;
    procedure setCode(ncode: integer);
    function getOrder(): integer;
    procedure setOrder(norder: integer);
    function getParentCodFolder(): integer;
    procedure setParentCodFolder(nparentCodFolder: integer);
    function getFolderType(): char;
    procedure setFolderType(sfolderType: char);
    function getPath(): String;
    procedure setPath(spath: String);
    function getParentName(): String;
    procedure setParentName(sparentName: String);
    function getParentPath(): String;
    procedure setParentPath(sparentPath: String);

    procedure limparDados;
    function ToString: ansistring; override;
    function ToInsert(naba: integer): string;
    function ToCVS: string;
    function ToJSON: string;
  end;

implementation

constructor TFolder.Create(prefile: TPreFile);
begin
  inherited Create;
  self.setName(preFile.getName());
  self.setSize(preFile.getSize());
  self.setModified(preFile.getModified());
  self.setAttributes(preFile.getAttributes());
  self.setFormatedSize(preFile.getFormatedSize());
  self.setFormatedModified(preFile.getFormatedModified());
  self.setOriginalPath(prefile.getOriginalPath());
  self.directory:=prefile.directory;
end;

destructor TFolder.Destroy;
begin
  inherited Destroy;
end;

function TFolder.getCode(): integer;
begin
  result:=self.code;
end;

procedure TFolder.setCode(ncode: integer);
begin
  self.code:=ncode;
end;

function TFolder.getOrder(): integer;
begin
   result:=self.order;
end;

procedure TFolder.setOrder(norder: integer);
begin
  self.order:=norder;
end;

function TFolder.getParentCodFolder(): integer;
begin
   result:=self.parentCodFolder;
end;

procedure TFolder.setParentCodFolder(nparentCodFolder: integer);
begin
  self.parentCodFolder:=nparentCodFolder;
end;

function TFolder.getFolderType(): char;
begin
   result:=self.folderType;
end;

procedure TFolder.setFolderType(sfolderType: char);
begin
  self.folderType:=sfolderType;
end;

function TFolder.getPath(): String;
begin
   result:=self.path;
end;

procedure TFolder.setPath(spath: String);
begin
   self.path:=spath;
end;

function TFolder.getParentName(): String;
begin
   result:=self.parentName;
end;

procedure TFolder.setParentName(sparentName: String);
begin
  self.parentName:=sparentName;
end;

function TFolder.getParentPath(): String;
begin
   result:=self.parentPath;
end;

procedure TFolder.setParentPath(sparentPath: String);
begin
  self.parentPath:=sparentPath;
end;


procedure TFolder.limparDados;
begin
  self.limparDados();
  self.code := 0;
  self.order := 0;
  self.parentCodFolder := 0;
  self.folderType := 'A';
  self.path := '';
  self.parentName := '';
  self.parentPath := '';
end;

function TFolder.ToString: ansistring;
begin
  inherited ToString;

  result:='Folder [codigo=' + IntToStr(code) + ', ordem=' + IntToStr(order) +
  ', codDirPai=' + IntToStr(parentCodFolder) + ', tipo=' + folderType +
  ', caminho=' + path + ', nomePai=' + parentName +
  ', caminhoPai=' + parentPath + ']';
end;

function TFolder.ToInsert(naba: integer): string;
begin
  result:='INSERT INTO Diretorios(aba, cod, ordem, nome, '+
  'tam, tipo, modificado, atributos, coddirpai, caminho) VALUES('
  + IntToStr(naba) + ',' + IntToStr(code) + ',' + IntToStr(order) + ',' +
  QuotedStr(getName()) + ',' + IntToStr(getSize()) + ',' +
  QuotedStr(folderType) + ',' + QuotedStr(getFormatedModified()) + ',' +
  QuotedStr(getAttributes()) + ',' + IntToStr(parentCodFolder) + ',' +
  QuotedStr(getPath()) + ');';
end;

function TFolder.ToCVS: string;
begin
  result:=IntToStr(code) + ';' + IntToStr(order) + ';' + getName() + ';' +
  IntToStr(getSize()) + ';' + folderType + ';' + getFormatedModified() + ';' +
  getAttributes() + ';' + IntToStr(parentCodFolder) + ';' + getPath();
end;

function TFolder.ToJSON: string;
begin
  result:= '{'+ LineEnding + '"name" : "' + self.getName() + '",' +
LineEnding + '  "size" : ' + IntToStr(getSize()) + ',' +
LineEnding + '  "modified" : ' + DateTimeToStr(self.getModified()) + ',' +
LineEnding + '  "attributes" : "' + self.getAttributes() + '",' +
LineEnding + '  "formatedSize" : "' + self.getFormatedSize() + '",' +
LineEnding + '  "formatedModified" : "' + self.getFormatedModified() + '",' +
LineEnding + '  "code" : ' + IntToStr(code) + ',' +
LineEnding + '  "order" : ' + IntToStr(order) + ',' +
LineEnding + '  "parentCodFolder" : ' + IntToStr(parentCodFolder) + ',' +
LineEnding + '  "folderType" : "' + self.folderType + '",' +
LineEnding + '  "path" : "' + self.path + '",' +
LineEnding + '  "parentName" : "' + self.parentName + '",' +
LineEnding + '  "parentPath" : "' + self.parentPath + '",' +
LineEnding + '  "originalPath" : "' + self.getOriginalPath() + '"\n}';
end;

end.

